File: solution/01xxx/1316 - 그룹 단어 체커/1316.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

#define FASTIO cin.tie(NULL); cout.tie(NULL); ios::sync_with_stdio(false);
#define MAX 33

int main() {
    FASTIO

    int n, result = 0;

    cin >> n;

    for(int _ = 0; _ < n; _++) {
        string word;
        vector<bool> count(MAX, false);
        bool stop = true;

        cin >> word;

        int word_size = word.size();

        for(int i = 0; i < word_size && stop; i++) {
            if(count[word[i]-'a'] == false) count[word[i]-'a'] = true;
            else if(word[i]-'a' != word[i-1]-'a') stop = false;
        }

        if(stop == true) result += 1;
    }

    cout << result;
}

File: solution/01xxx/1316 - 그룹 단어 체커/1316.py
import sys
input = sys.stdin.readline

def check(word):
    count = [0 for _ in range(26)]

    for i in range(len(word)):
        if count[ord(word[i]) - ord('a')] == 0:
            count[ord(word[i]) - ord('a')] += 1
        elif count[ord(word[i]) - ord('a')] != 0 and word[i] != word[i-1]:
            return 0
    
    return 1

n = int(input())
result = 0

for _ in range(n):
    word = input()[:-1]
    result += check(word)

print(result)